tests/__init__.py

```diff
@@ -359,3 +361,64 @@ def assert_url_equal(url1, url2):
     assert_equal(parts1.hostname, parts2.hostname)
     assert_equal(parts1.port, parts2.port)
     assert_equal(parse_qs(parts1.query), parse_qs(parts2.query))
+
+
+class HTTPStubberException(Exception):
+    pass
+
+
+class RawResponse(BytesIO):
+    def stream(self, **kwargs):
+        contents = self.read()
+        while contents:
+            yield contents
+            contents = self.read()
+
+
+class ClientHTTPStubber(object):
+    def __init__(self, client):
+        self.reset()
+        self._client = client
+
+    def reset(self):
+        self.requests = []
+        self.responses = []
+
+    def create_response(self, url=None, status=200, headers=None, body=None):
+        if url is None:
+            url = 'https://example.com'
+
+        if headers is None:
+            headers = {}
+
+        if body is None:
+            body = b''
+
+        raw = RawResponse(body)
+        response = AWSResponse(url, status, headers, raw)
+        self.responses.append(response)
+
+    def start(self):
+        self._client.meta.events.register('before-send', self)
+
+    def stop(self):
+        self._client.meta.events.unregister('before-send', self)
+
+    def __enter__(self):
+        self.start()
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.stop()
+
+    def __call__(self, **kwargs):
+        request = kwargs.get('request')
```

Comment:
I don't follow when `request` wouldn't be in the kwargs.  Why not use `__call__(self, request, **kwargs)`.

Reply:
It will always be there, not sure why I'm doing the get to be honest.